File: php/features.php
<?php
declare(strict_types=1);

// AffirmationGenerator SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AffirmationGeneratorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AffirmationGeneratorBaseFeature();
            case "test":
                return new AffirmationGeneratorTestFeature();
            default:
                return new AffirmationGeneratorBaseFeature();
        }
    }
}

File: py/features.py
# AffirmationGenerator SDK feature factory

from feature.base_feature import AffirmationGeneratorBaseFeature
from feature.test_feature import AffirmationGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AffirmationGeneratorBaseFeature(),
        "test": lambda: AffirmationGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
